app/models/group.rb

```diff
@@ -105,6 +105,7 @@ class MaximumMembershipsExceeded < Exception
   has_many :memberships,
            -> { where is_suspended: false, archived_at: nil },
            extend: GroupMemberships
+  define_counter_cache(:memberships_count) { |group| group.memberships.count }
```

Comment:
This moves the N+1 problem from every time we load the Group to everytime we update the Group... which is better but still not the best.

I'm seriously considering writing a gem to handle this with increment / decrement.